api/src/main/java/org/apache/cloudstack/api/command/user/vm/DeployVMCmd.java

```diff
@@ -765,28 +768,33 @@ public ApiCommandResourceType getApiResourceType() {
     public void execute() {
         UserVm result;
 
-        try {
-            CallContext.current().setEventDetails("Vm Id: " + getEntityUuid());
-            result = _userVmService.startVirtualMachine(this);
-        } catch (ResourceUnavailableException ex) {
-            s_logger.warn("Exception: ", ex);
-            throw new ServerApiException(ApiErrorCode.RESOURCE_UNAVAILABLE_ERROR, ex.getMessage());
-        } catch (ResourceAllocationException ex) {
-            s_logger.warn("Exception: ", ex);
-            throw new ServerApiException(ApiErrorCode.RESOURCE_ALLOCATION_ERROR, ex.getMessage());
-        } catch (ConcurrentOperationException ex) {
-            s_logger.warn("Exception: ", ex);
-            throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, ex.getMessage());
-        } catch (InsufficientCapacityException ex) {
-            StringBuilder message = new StringBuilder(ex.getMessage());
-            if (ex instanceof InsufficientServerCapacityException) {
-                if (((InsufficientServerCapacityException)ex).isAffinityApplied()) {
-                    message.append(", Please check the affinity groups provided, there may not be sufficient capacity to follow them");
+        CallContext.current().setEventDetails("Vm Id: " + getEntityUuid());
```

Comment:
As my comment on https://github.com/apache/cloudstack/pull/7714#discussion_r1251610251

I have not seen this done in Cmd classes yet. Usually it happens in the service, i think. Did you consider that as well? (and what were the pros and cons)

It would be good to keep such considerations for posterity.

Reply:
In this case there is a call made from the command to API: _userVmService.startVirtualMachine(this);
startVirtualMachine(..) intercepted by ActionEventInterceptor, which needs entity Id to add it to the event.
If startVirtualMachine(..) called by API layer, entity Id set to the context from the filter and there is no issue, but if it is called from command or any other layer, which may not have entity Id, then Id is missing in the event and event itself does not point to the right resource.
This is why it is set explicitly before making call to the API layer.
As of now I don't see other simple enough way how to add Id to the context.